src/main/scala/org/ergoplatform/nodeView/ErgoNodeViewHolder.scala

```diff
@@ -37,6 +39,15 @@ abstract class ErgoNodeViewHolder[State <: ErgoState[State]](settings: ErgoSetti
   override protected lazy val modifiersCache =
     new ErgoModifiersCache(settings.scorexSettings.network.maxModifiersCacheSize)
 
+  override def preStart(): Unit = {
+    // enable periodic mempool cleanup for non-mining regimes
+    if (!settings.nodeSettings.mining) {
```

Comment:
For mining regimes it might make sense, especially for initial initialization. Or we can just disable mempool for initialization time